Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs

```diff
@@ -0,0 +1,727 @@
+    }
+    /// <summary>
+    /// Selected item
+    /// </summary>
+    public IList<object> SelectedItems
+    {
+        get => (IList<object>)GetValue(SelectedItemsProperty);
+        set => SetValue(SelectedItemsProperty, value);
+    }
+    /// <summary>
+    /// Column index and sorting order for the DataGrid
+    /// </summary>
+    public SortData SortedColumnIndex
+    {
+        get => (SortData)GetValue(SortedColumnIndexProperty);
+        set => SetValue(SortedColumnIndexProperty, value);
+    }
+
+    /// <summary>
+    /// Sort icon
+    /// </summary>
+    public Polygon SortIcon
+    {
+        get => (Polygon)GetValue(SortIconProperty);
+        set => SetValue(SortIconProperty, value);
+    }
+
+    /// <summary>
+    /// View to show when there is no data to display
+    /// </summary>
+    public View NoDataView
+    {
+        get => (View)GetValue(NoDataViewProperty);
+        set => SetValue(NoDataViewProperty, value);
+    }
+
+    #endregion Properties
+
+    #region UI Methods
+
+    /// <inheritdoc/>
+    protected override void OnParentSet()
+    {
+        base.OnParentSet();
+
+        if (SelectionMode != SelectionMode.None)
+        {
+            if (Parent is null)
+            {
+                _collectionView.SelectionChanged -= OnSelectionChanged;
+            }
+            else
+            {
+                _collectionView.SelectionChanged += OnSelectionChanged;
+            }
+        }
+
+        if (Parent is null)
+        {
+            Columns.CollectionChanged -= OnColumnsChanged;
+
+            foreach (var column in Columns)
+            {
+                column.SizeChanged -= OnColumnSizeChanged;
+            }
+        }
+        else
+        {
+            Columns.CollectionChanged += OnColumnsChanged;
+
+            foreach (var column in Columns)
+            {
+                column.SizeChanged += OnColumnSizeChanged;
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+    protected override void OnBindingContextChanged()
+    {
+        base.OnBindingContextChanged();
+        InitHeaderView();
+    }
+
+    private void OnColumnsChanged(object? sender, NotifyCollectionChangedEventArgs e) => Reload();
+
+    private void OnColumnSizeChanged(object? sender, EventArgs e) => Reload();
+
+    private void OnSelectionChanged(object? sender, Microsoft.Maui.Controls.SelectionChangedEventArgs e)
+    {
+        SelectedItem = SelectionMode != SelectionMode.None? _collectionView.SelectedItem : null;
+        SelectedItems =  _collectionView.SelectedItems;
+        _itemSelectedEventManager.HandleEvent(this, e, nameof(ItemSelected));
+
+    }
+
+    /// <summary>
+    /// Reload the table
+    /// </summary>
+    internal void Reload()
+    {
+        InitHeaderView();
+        _collectionView.BackgroundColor = BorderColor;
+
+        if (_internalItems is not null)
+        {
+            InternalItems = new List<object>(_internalItems);
```

Comment:
Not sure if I get this right. _internalItems gets updated whenever the setter of InternalItems is updated, right? So why is this needed?

Reply:
We are assigning the items in `_internalItems` to `InternalItems` so that it will be considered as new list of items and the rows will reloaded.